docs/configuring.md

```diff
@@ -1035,6 +1036,151 @@ _commit: v1.0.0
+maintain.
+
+As it happens, you find that to simplify your template, you would like to be able to add
+or change variables of the Jinja context used to render templated files and folders, as
+it would allow to get rid of some questions that would then become redundant, and would
+not need to be asked to the users anymore.
+
+An example was brought up by a Copier user. Let say you support multiple flavors of
+containerization for Terraform projects generated with your template:
+
+```yaml title="copier.yaml"
+flavor:
+    type: str
+    choices:
+        - Docker
+        - Instances
+        - Kubernetes
+        - None
+```
+
+You generate different files depending on users answers:
+
+```
+.
+├── {% if flavor == 'docker' %}ecs{% endif %}
+├── {% if flavor == 'kubernetes' %}eks{% endif %}
+├── {% if flavor == 'instance' %}bastion{% endif %}
+├── {% if flavor == 'docker' or flavor == 'kubernetes' %}portainer{% endif %}
+└── {% if flavor != 'none' %}vpc{% endif %}
+```
+
+Each of these folders then have multiple templated files that must also repeat again and
+again these Jinja conditions to render the contents correctly.
+
+You can reduce repetitions with
+[macros](https://jinja.palletsprojects.com/en/2.10.x/templates/#macros) or updated
+context objects that you can import from special templated files
+([see issue #229](https://github.com/copier-org/copier/issues/229#issuecomment-661686740)).
+
+It would be easier if you could add variables to the context, like `isDocker`, or
+`hasContainers` when either Docker or Kubernetes was selected. It would make it easier
+to write your templated files and folders since you would just have to use the added
+variables instead of potentially complex Jinja conditions:
+
+```
+.
+├── {% if isDocker %}ecs{% endif %}
+├── {% if isKubernetes %}eks{% endif %}
+├── {% if isInstance %}bastion{% endif %}
+├── {% if hasContainers %}portainer{% endif %}
+└── {% if isLite %}vpc{% endif %}
+```
+
+We have good news for you: this is possible. A Copier/Jinja extension was written
+specially for this use-case:
+[copier-templates-extensions](https://github.com/copier-org/copier-templates-extensions)
+and its
+[`ContextHook` extension](https://github.com/copier-org/copier-templates-extensions#context-hook-extension).
+
+### Using a context hook
+
+The
+[`ContextHook` extension](https://github.com/copier-org/copier-templates-extensions#context-hook-extension)
+lets you modify the context used to render templates, so that you can add, change or
+remove variables.
+
+In order for Copier to be able to load and use the extension when generating a project,
+it must be installed alongside Copier itself. If Copier was installed with pip in a
+virtualenv, just `pip install copier-templates-extensions` as well, and tell your users
+to do the same. If Copier was installed with
+[pipx](https://github.com/pipx-project/pipx), inject it with
+`pipx inject copier copier-templates-extensions`, and tell your users to do the same.
```

Comment:
This paragraph is repeated above in the jinja extensions section. Maybe just link to it? Unless you move these docs to the extension.

Reply:
Ah, yes, we can probably just point to it then.